app/decorators/models/solidus_related_products/variant/add_relation_methods.rb

```diff
@@ -10,37 +10,30 @@ def self.prepended(base)
 
         # When a Spree::Variant is destroyed, we also want to destroy all Spree::Relations
         # "from" it as well as "to" it.
-        base.after_discard :destroy_variant_relations if base.respond_to?(:after_discard)
-        base.after_destroy :destroy_variant_relations
+        base.after_discard :destroy_variant_relations_to if base.respond_to?(:after_discard)
```

Comment:
I'm not convinced about this change. First, it's not a refactor as the commit message says, but a functional change. The comment above says that we need to destroy all relations, both from and to, and here we are only destroying the to relations? Not saying it's wrong but I want to understand why, and again, these kinds of changes deserve their own commits with a detailed explanation of the reasons driving them.

Please split these changes as much as possible for an easier review.

Reply:
@kennyadsl can you look again?